### DISCUSSION TITLE
Storage Page

### DISCUSSION THREAD

**Discussion by NickCondon** [UQ-RCC/ipp#73] [Ideas]
We should look to add a top level page (like Files Manager etc) for "Storage". This page would pull in information from RDM and initially Wiener but possibly in the future using the DMF7? to know where else caches are mounted and what resources they are using.

Ideally we shuoldnt limit this to just RDM Collections but also scratch spaces. We could then implement here the flush scratch either at the whole level (ie all my files) or at sub levels (.IPP folder, CVL/<USR> or ORG/<USR>).

This would also have the benefit of helping users not blow over their allocations.

Example layout below
********************** 

Storage Page

At Portal login query run to all locations to populate the page OR queries DMF7 for details...

[Refresh Stats] 	(Last update: hh:mm dd/mm/yy)

Storage @QBI (Wiener) (Expandable dropdown)
	collections
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage	[RDM Management Link]

	[Request a new Cache Mount @ QBI](emailto: help@qbi....  : Please mount X...)

	Scratch (2TB) 	+ Fuel Gage
		- .IPP				Usage		Clear
		- CVL				Usage		Clear
		- IMB				Usage		Clear

	Home Storage (10GB)
		- Desktop 			Usage		Clear
	

	

Storage @IMB	(Expandable dropdown)
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]

	[Request a new Cache Mount @ IMB](emailto: help@IMB....  : Please mount X...)

	Scrach
		- IMB Microscopy Scatch		Usage		Clear
		- Delta Scatch			                Usage		Clear	

Storage @AIBN  (expandable dropdown)
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]
		- XXXXX - QXXXX 	usage (Fuel Gauge)	request more storage 	[RDM Management Link]

	
	[Request a new Cache Mount @ AIBN](emailto: help@aibn....  : Please mount X...)
	
	Scrach
		- IMB Microscopy Scatch		Usage		Clear
		- Delta Scatch		 	                Usage		Clear	

Compare my multi institute mounted collections (RDMs)	OR for each cache above have a compare button.
(if mounted at more than one location...)

RDM Name		IMB		QBI		AIBN
XXXX-Q2XXX (5TB)	400GB/4TB	2TB/4TB		0/0TB
XXXX-Q45XX (10TB)	3TB/5TB		6TB/10TB	0/0TB

***************
Notes: RDM Management link appears to be rdm.uq.edu.au/record/shortcode so should be pretty scriptable....

**Comment by 7neyugn7** [comment]
I think this is a great idea. But it seems to be more than IPP scope. For me, it is more like a MeDiCi feature, don't you think ? I will ask others to see whether they have smth similar.

**Comment by DrSpringy** [comment]
I think this is definitely within the scope of the project as it allows a user to interact with complex file stores in a simple intuitive and limited fashion.
If they need further detail they can click on the link and go to the appropriate RDM, QCIF etc website
This is basically just a fuel gauge for each collection sorted by location to let users identify issues and view all of their collections that they use on the portal in one spot, which is on the portal
